File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int product=1, zeroCount=0;

        for(int num : nums){
            if(num!=0){
                product*=num;
            }else{
                zeroCount++;
            }
        }

        if(zeroCount>1){
            return vector<int>(nums.size(), 0);
        }

        vector<int> res(nums.size());
        for(int i=0; i<res.size(); i++){
            if(zeroCount>0){
                res[i] = (nums[i]==0) ? product:0;
            }else{
                res[i] = product/nums[i];
            }
        }
    return res;
        
    }
};

File: Data Structures & Algorithms/counting-bits/submission-0.py
class Solution:
    def count(self, i):
        count = 0 
        while i > 0:
            if i & 1 == 1:
                count += 1
            i = i >> 1
        return count 

    def countBits(self, n: int) -> List[int]:
        res = []
        for i in range(n + 1):    
            res.append(self.count(i))
        return res
        